nntrainer/src/tensor.cpp

```diff
@@ -63,15 +63,15 @@ static auto rng = [] {
 
 Tensor::Tensor(const TensorDim d) {
   dim = d;
-  this->data = std::vector<float>(dim.getDataLen());
+  this->_data = std::make_shared<std::vector<float>>(dim.getDataLen());
```

Comment:
This is not related to this PR but I think you can declare and use a base constructor of `Tensor` initializing common variables.

Reply:
That is actually a good idea. Would do in another PR. Thanks for the suggestion!